bin/bootstrap_practice_exercise.exs

```diff
@@ -0,0 +1,258 @@
+# Generate all files required for a practice exercise.
+# File content is filled as much as possible, but some, including tests, need manual input.
+#
+# Run the following command from the root of the repo:
+# $ elixir bin/boostrap_practice_exercise.exs complex-numbers
+# Pass the name of the exercise (e. g., "complex-numbers") as an argument
+
+defmodule Generate do
+  def to_snake_case(lower_camel_case) do
+    lower_camel_case
+    |> String.split(~r/(?=[A-Z\d])/)
+    |> Enum.map_join("_", &String.downcase/1)
+  end
+
+  def get_properties(%{"cases" => cases}) when is_list(cases) do
+    Enum.map(cases, &Generate.get_properties/1)
+    |> Enum.concat()
+    |> Enum.uniq()
+  end
+
+  def get_properties(%{"property" => property, "input" => %{} = input}),
+    do: [
+      {property,
+       Enum.map(input, fn
+         {var, %{} = val} -> {var, Enum.map(val, fn {v, _} -> v end)}
+         {var, _} -> {var, nil}
+       end)}
+    ]
+
+  def get_properties(%{"property" => property}), do: [{property, "input"}]
+  def get_properties(_data), do: []
+
+  def print_property({property, variables}) do
+    """
+     @doc \"\"\"
+     insert function description here
+     \"\"\"
+     @spec #{property}(#{Enum.map_join(variables, ", ", fn
+      {var, nil} -> "#{var} :: String.t()"
+      {var, sub_vars} -> "#{var} :: %{#{Enum.map_join(sub_vars, " : String.t(), ", fn v -> v end)} : String.t()}"
+    end)}) :: {:ok, String.t()} | {:error, String.t()}
+     def #{property}(#{Enum.map_join(variables, ", ", fn {var, _} -> var end)}) do
+     end
+    """
+  end
+
+  def print_comments(comments, do_not_print) do
+    comments
+    |> Enum.reject(fn {field, _value} -> field in do_not_print end)
+    |> Enum.map_join("\n", fn
+      {"comments", values} when is_list(values) ->
+        "# #{Enum.map_join(values, "\n# ", &String.trim/1)}"
+
+      {field, values} when is_list(values) ->
+        "#\n# --#{field} --\n# #{Enum.map_join(values, "\n# ", &inspect/1)}"
+
+      {field, value} ->
+        "#\n# -- #{field} --\n# #{inspect(value)}"
+    end)
+  end
+
+  def print_input(%{} = input) do
+    Enum.map_join(input, "\n", fn {variable, value} -> "#{variable} = #{inspect(value)}" end)
+  end
+
+  def print_input(input), do: "input = #{inspect(input)}"
+
+  def print_variables(%{} = input),
+    do: Enum.map_join(input, ", ", fn {variable, _value} -> variable end)
+
+  def print_variables(input), do: input
+
+  def print_expected(%{"error" => err}), do: "{:error, #{inspect(err)}}"
+  def print_expected(expected), do: "{:ok, #{inspect(expected)}}"
```

Comment:
Would it be possible (and not too difficult) to make this more intelligent, and only use `{:ok, value}` if any error case exists in the problem specification for this property, and otherwise just use the plain value? There are a lot of exercises without any error handling.

Reply:
Done :)